vulcan/lib/client/jsx/api_types.ts

```diff
@@ -102,7 +102,7 @@ export interface Workflow {
   steps: [WorkflowStep[]];
   vignette?: string;
   image?: string;
-  projects?: string[];
+  projects?: string[] | null;
```

Comment:
:+1: 
Interesting aside, `projects?` implies `projects | undefined`, specifically also that the `projects` key is possibly non existent.

It's a bit of a mind bender, but recall that in javascript there are 3 "empty" states for an object property.
1.  The property exists and is the `undefined` value.  (Generally meant to signal the explicit desire to 'remove' a key)
2.  The property exists and is the `null` value.  (Generally meant to signal the explicit desire to 'keep' a key as an empty value)
3.  The property does not exist, and is `undefined` when accessed.  (Generally meant to signal the explicit lack of knowledge of the key).

:exploding_head:

Reply:
yes... I stopped short, but I also think we could make this a non-optional property.  Just probably would involve updating a bunch of tests and maybe other real code too.  I do think we'll want workflow maintainers to always give an explicit projects scope.

Reply:
Thing is, all the other required bits of the Wrokflow type come from the CWL (except `class`. idk what that is.) so I'm not sure if I should remove the `?` or not.